### PR TITLE
Tokenizer + AddController parsing

### LINKED ISSUE #31: Flexi-commands for parser
We need to able to re-order the input by the user and still be able to pass to the right parser.
We need to support start date and end date

Format : 
`add task1 by today`
`add by today task 2`
`add lesson from 2pm to 4pm`
`add v0.3 from today to friday`

### PR DESCRIPTION
**Pull request by louietyj**
Fixes #31

### CODE REVIEW

**Comment by louietyj** [comment]
Okay yes my cyclomatic complexity is off the charts, gotta fix that in a separate PR.

**Comment by ChaseTiong** [comment]
Need to support for floating task. I think we should either set it today as natural date or use null to display as no deadline

**Comment by louietyj** [comment]
This PR is for parsing `add` commands, your floating task thing is a separate thing. Can create issue.

Your original code to set today as default should not be the solution regardless.

**Comment by louietyj** [comment]
Fixes #31

File: src/main/java/seedu/todo/controllers/AddController.java [hunk 1]
```diff
@@ -31,79 +35,102 @@ public float inputConfidence(String input) {
...
-            }
-        }
-        
-        return Math.max(indexOfBy, indexOfOn);
-        
+    private static Map<String, String[]> getTokenDefinitions() {
+        Map<String, String[]> tokenDefinitions = new HashMap<String, String[]>();
+        tokenDefinitions.put("default", new String[] {"add"});
+        tokenDefinitions.put("eventType", new String[] { "event", "task" });
+        tokenDefinitions.put("time", new String[] { "at", "by", "before", "time" });
```
Comment (irvinlim) [inline src/main/java/seedu/todo/controllers/AddController.java]: Can we add a "on" token for dates? E.g. `add CS2103 Tutorial on Friday`

**Comment by irvinlim** [comment]
Can I clarify if the tokenizer will, in the event of an ambiguous command like `add Buy fruits at NTUC at 3pm` will prompt the user for disambiguation? Because right now it's parsing it into simply "Buy fruits" without prompt.

**Comment by louietyj** [comment]
@irvinlim current implementation doesn't detect problems like this on the tokenizer side, but for duplicate tokens it is conceivable that we throw an error when we notice it. But as for how to disambiguate a duplicate column... I'm not entirely sure because the tokenizer will present two "at"s.

**Comment by louietyj** [comment]
It also requires a seriously more complicated return data structure to be able to support returning multiple detected tokens. Right now it's still a single hashmap because I'm cutting corners by assuming only one match per tokenType.

**Pull request merged**